addons/message_update_v2.12/update.js

```diff
@@ -1,3 +1,34 @@
 ((api) => {
-  api.navigator.requestScreen(api.navigator.ScreenUpdateRecommended);
+  // windows v2.10/v2.11 do require a web-based update.
+  if (api.env.platform != 'windows') {
+    api.navigator.requestScreen(api.navigator.ScreenUpdateRecommended);
+    return;
+  }
+
+  const parts = api.env.versionString.split('.');
+
+  // No idea which version we are in...
+  if (parts.length < 3) {
+    api.navigator.requestScreen(api.navigator.ScreenUpdateRecommended);
+    return;
+  }
+
+  const version = parts.map(a => parseInt(a, 10));
+
+  function versionCompare(a, b) {
+    for (let i = 0; i < 3; ++i) {
+      if (a[i] != b[i]) {
+        return a[i] > b[i] ? -1 : 1;
+      }
+    }
+    return 0;
+  }
+
+  if (versionCompare([2, 11, 1], version) >= 0 ||
+      versionCompare([2, 10, 0], version) < -1) {
+    api.navigator.requestScreen(api.navigator.ScreenUpdateRecommended);
```

Comment:
In this case, the balrog code should be working just fine. Shouldn't we simply doing nothing and leave the update recommendation screen to Balrog?

Reply:
> I have mild trepidation that we are going to pop open a browser session directly to the 2.12 update page every time the client is opened. Is there a mechanism to dismiss this, users may find this to be somewhat jarring.

this script runs only when the user clicks the button. This is the CTA for the button, see the manifest.json.

Reply:
> In this case, the balrog code should be working just fine. Shouldn't we simply doing nothing and leave the update recommendation screen to Balrog?

Correct. This is what that screen does. That screen triggers balrog and the self-update.

Reply:
Excellent, that addresses my concerns.